scabha/evaluator.py

```diff
@@ -625,6 +625,31 @@ def evaluate_dict(self, params: Dict[str, Any],
                         params_out[name] = new_value
                         if corresponding_ns:
                             corresponding_ns[name] = new_value
+                elif isinstance(value, (dict, DictConfig)):
+                    params_out[name] = self.evaluate_dict(
+                        value,
+                        corresponding_ns,
+                        defaults,
+                        sublocation,
+                        raise_substitution_errors,
+                        recursive,
+                        verbose
+                    )
+                elif isinstance(value, list):
```

Comment:
@JSKenyon clever, clever. Maybe use ``(list, ListConfig)`` just in case?

Reply:
Even better:

```python
params_out[name] = type(value)([x for x in
    self.evaluate_dict({f"[{i}]": v for i, v in enumerate(value)}, ...).values()
   ])
```

...because then any nested error messages will report the location of the error sensibly as "[n]".

Reply:
Ah, awesome. Definitely better than a profusion of `dummies`!

Reply:
Actually no need for the extra list comprehension, right? This should work too: 

```python
type(value)(self.evaluate_dict({f"[{i}]": v for i, v in enumerate(value)}, ...).values())
```

Reply:
Ok, I have made a final change. Let me know if it appeals.

Reply:
Oh, also the ``evaluate_dict()`` call needs ``sublocation=sublocation + [name]``, so that errors are reported from the correct nested namespace.

Reply:
Thanks for making those changes @o-smirnov - was only going to get to it today.